core/trino-main/src/main/java/io/trino/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -121,6 +124,8 @@ public synchronized void stop()
             future.cancel(true);
             future = null;
         }
+        remoteTaskCleaner.markTaskStatusFetcherStopped(taskStatus.get().getState());
```

Comment:
I would postpone that only after pending request is finished, so that last DF update can be processed

Reply:
how is this one related to pending requst in DF fetcher?